Mathlib/CategoryTheory/Triangulated/TStructure/TruncLTGE.lean

```diff
@@ -0,0 +1,244 @@
+    simp [hg, t.zero_of_isLE_of_isGE g a (b + 1) (by cutsat)
+      h₀ (t.isGE_shift _ b (-1) (b + 1))]
+  · simp [hf]
+  · obtain ⟨g, hg⟩ := T.yoneda_exact₃ hT f.hom₃ (by cat_disch)
+    simp [hg, t.zero_of_isLE_of_isGE g (a - 1) b (by cutsat)
+      (t.isLE_shift _ a 1 (a - 1)) inferInstance]
+
+/-- If `a < b`, then a morphism `T.obj₂ ⟶ T'.obj₂` extends to a morphism `T ⟶ T'`
+of distinguished triangles when for a t-structure `T.obj₁ ≤ a` and `T'.obj₃ ≥ b`. -/
+lemma triangle_map_exists (a b : ℤ) (h : a < b) (T T' : Triangle C)
+    (hT : T ∈ distTriang C) (hT' : T' ∈ distTriang C)
+    (φ : T.obj₂ ⟶ T'.obj₂) (h₀ : t.IsLE T.obj₁ a) (h₁' : t.IsGE T'.obj₃ b) :
+    ∃ (f : T ⟶ T'), f.hom₂ = φ := by
+  obtain ⟨a, comm₁⟩ := T'.coyoneda_exact₂ hT' (T.mor₁ ≫ φ) (t.zero _ a b)
+  obtain ⟨c, ⟨comm₂, comm₃⟩⟩ := complete_distinguished_triangle_morphism _ _ hT hT' a φ comm₁
+  exact ⟨{ hom₁ := a, hom₂ := φ, hom₃ := c }, rfl⟩
+
+/-- If `a < b`, then an isomorphism `T.obj₂ ≅ T'.obj₂` extends to an morphism `T ≅ T'`
+of distinguished triangles when for a t-structure, both `T.obj₁` and `T'.obj₁` are `≤ a` and
+both `T.obj₃` and `T'.obj₃` are `≥ b`. -/
+lemma triangle_iso_exists (a b : ℤ) (h : a < b) (T T' : Triangle C)
+    (hT : T ∈ distTriang C) (hT' : T' ∈ distTriang C) (e : T.obj₂ ≅ T'.obj₂)
+    (h₀ : t.IsLE T.obj₁ a) (h₁ : t.IsGE T.obj₃ b)
+    (h₀' : t.IsLE T'.obj₁ a) (h₁' : t.IsGE T'.obj₃ b) :
+    ∃ (e' : T ≅ T'), e'.hom.hom₂ = e.hom := by
+  obtain ⟨hom, hhom⟩ := triangle_map_exists t _ _ h _ _ hT hT' e.hom h₀ h₁'
+  obtain ⟨inv, hinv⟩ := triangle_map_exists t _ _ h _ _ hT' hT e.inv h₀' h₁
+  exact ⟨
+    { hom := hom
+      inv := inv
+      hom_inv_id := triangle_map_ext t a b (by cutsat) _ _ hT hT h₀ h₁
+        (by cat_disch)
+      inv_hom_id := triangle_map_ext t a b (by cutsat) _ _ hT' hT' h₀' h₁'
+        (by cat_disch) }, hhom⟩
+
+namespace TruncAux
+/-! The definitions in the namespace `TStructure.TruncAux` are part of the
+implementation of the truncation functors `truncLT`, `truncGE` and the
+distinguished triangles they fit in. -/
+
+variable (n : ℤ) (X : C)
+
+/-- Given a t-structure `t` on `C`, `X : C` and `n : ℤ`, this is a distinguished
+triangle `obj₁ ⟶ X ⟶ obj₃ ⟶ obj₁⟦1⟧` where `obj₁` is `< n` and `obj₃` is `≥ n`.
+(This should not be used directly: use `truncLT` and `truncGE` instead.) -/
+@[simps! obj₂]
+noncomputable def triangle : Triangle C :=
+  Triangle.mk
+    (t.exists_triangle X (n-1) n
+      (by cutsat)).choose_spec.choose_spec.choose_spec.choose_spec.choose
+    (t.exists_triangle X (n-1) n
+      (by cutsat)).choose_spec.choose_spec.choose_spec.choose_spec.choose_spec.choose
+    (t.exists_triangle X (n-1) n
+      (by cutsat)).choose_spec.choose_spec.choose_spec.choose_spec.choose_spec.choose_spec.choose
+
+lemma triangle_distinguished :
+    triangle t n X ∈ distTriang C :=
+  (t.exists_triangle X (n-1) n (by cutsat)
+    ).choose_spec.choose_spec.choose_spec.choose_spec.choose_spec.choose_spec.choose_spec
```

Comment:
```suggestion
  (t.exists_triangle X (n-1) n
    (by cutsat)).choose_spec.choose_spec.choose_spec.choose_spec.choose_spec.choose_spec.choose_spec
```
thankfully that last line is 100 characters long and no more, and avoids the `    ).`, I have to admit this is a weird situation.